apps/04TowersOfHanoi.js

```diff
@@ -16,24 +16,64 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(startStack, endStack) {
     // Your code here
+    // pop
+    var piece = stacks[startStack].pop();
+    // push
+    stacks[endStack].push(piece);
 
 }
 
-function isLegal() {
+function isLegal(startStack, endStack) {
     // Your code here
+    var startStackArray = stacks[startStack];
+    var endStackArray = stacks[endStack];
+      //var lastElementOfStart = startStackArray[startStackArray.lenghth - 1];
+      //  var lastElementofEnd = endStack[endStack.lenghth - 1];
+    if (stacks[startStack].length === 0){
```

Comment:
Good logic, but you have a variable that stores the array already (no need to go get it from the object again)